### PR TITLE
Provide device presets in separate files

### PR DESCRIPTION
**Pull request by HoLo85**
This PR moves the presets into separate files in the new `sma-device-presets/` folder. The presets are read when the node starts and are provided as before, accessing the presets does not change.

The presets have to be placed in a file named like the options in the device selection, e.g. `sb_tripower.json` and provide the following data:
```
{
    "name": "SUNNY TRIPOWER 8.0/10.0",
    "config": {
        "id": "1",
        "values": {
            "6100_0046E500": {
                "name": "phase1_voltage",
                "divider": 100
            },
            "6100_0046E600": {
                "name": "phase2_voltage",
                "divider": 100
            },
            "6100_0046E700": {
                "name": "phase3_voltage",
                "divider": 100
            },
            "6100_40463600": {
                "name": "grid_feedin",
                "divider": 1
            },
            "6100_40463700": {
                "name": "grid_consumption",
                "divider": 1
            },
            "6100_40263F00": {
                "name": "power",
                "divider": 1
            }
        }
    }
}
```

### CODE REVIEW

**Review by rlindner — CHANGES_REQUESTED** [review]
Thanks for your proposed changes! Nice improvement! One thing I noticed is that the node wasn't working after packaging. The `package.json` includes a `files` array, where you need to add the newly added directory. I'd suggest to add the whole directory with all files in it with `sma-device-presets/*` (See another comment about the name of that directory). You can test it by creating a package with `npm pack`.

File: sma-device-presets/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    },
+
+    getList: () => {
+        return devList;
+    },
+
+    getConfigs: () => {
+        return devConfigs;
+    }
+};
```
[the remark this answers is not in the document]
Comment (HoLo85) [reply sma-device-presets/index.js:26]: I would like to keep them for future use. The idea is to later update the device list in the web interface automatically without adding the devices manually.

**Comment by HoLo85** [comment]
> 
> 
> Thanks for your proposed changes! Nice improvement! One thing I noticed is that the node wasn't working after packaging. The `package.json` includes a `files` array, where you need to add the newly added directory. I'd suggest to add the whole directory with all files in it with `sma-device-presets/*` (See another comment about the name of that directory). You can test it by creating a package with `npm pack`.

Sorry, forgot that. The folder and files are now included in the package.

File: sma-device-presets/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (rlindner) [reply sma-device-presets/index.js]: Then I'd prefer you keep those changes in your local checkout/branch or your fork and you add them here again together with the functionality which requires it. Code which isn't used in any way is prone to be removed.

**Comment by HoLo85** [comment]
@rlindner did you have a look at the last changes and can we merge the PR?

**Comment by rlindner** [comment]
There's still a reference to `devList` which prevents to node from loading:

```
ReferenceError: devList is not defined (line:8)
```

Once that's resolved, I'm fine with merging it. 👍

**Pull request merged**